packages/amplify-appsync-simulator/src/server/index.ts

```diff
@@ -17,10 +19,30 @@ export class AppSyncSimulatorServer {
   private _realTimeSubscriptionServer: AppSyncSimulatorSubscriptionServer;
   private _url: string;
   private _localhostUrl: string;
+  private _isHttps = false;
 
   constructor(private config: AppSyncSimulatorServerConfig, private simulatorContext: AmplifyAppSyncSimulator) {
     this._operationServer = new OperationServer(config, simulatorContext);
-    this._httpServer = createServer(this._operationServer.app);
+
+    // Check if the SSL key path and certificate path exist and are valid
+    if (!config.sslKeyPath || !config.sslCertPath) {
+      this._httpServer = createServer(this._operationServer.app);
+    } else {
```

Comment:
The property that enables https should be more explicit instead of using ssl key/cert properties. 
Please see my comment in API.md file for suggestion how to make it better.